src/lib/message-utils.ts

```
@@ -0,0 +1,227 @@
+  }
+
+  // Validate npx commands with whitelist
+  if (normalizedCommand.startsWith('npx ')) {
+    const packageMatch = command.match(/^npx\s+(@?[\w@\/.-]+)(?:\s|$)/);
+    if (!packageMatch) {
+      return { isValid: false, error: 'Invalid npx command format' };
+    }
+    
+    const packageName = packageMatch[1];
+    const isApproved = APPROVED_MCP_PACKAGES.some(approved => 
+      packageName === approved || packageName.startsWith(approved + '@'),
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)